sui/tests/full_node_tests.rs

```diff
@@ -40,6 +54,61 @@ async fn transfer_coin(
     Ok((object_to_send, sender, receiver, digest))
 }
 
+async fn wait_for_tx(wait_digest: TransactionDigest, state: Arc<AuthorityState>) {
+    let mut timeout = Box::pin(sleep(Duration::from_millis(5000)));
+
+    let mut max_seq = Some(0);
+
+    let mut stream = Box::pin(
+        state
+            .handle_batch_streaming(BatchInfoRequest {
+                start: max_seq,
+                length: 1000,
+            })
+            .await
+            .unwrap(),
+    );
+
+    loop {
+        tokio::select! {
+            _ = &mut timeout => panic!("wait_for_tx timed out"),
```

Comment:
Instead of using this paradigm would it be possible to use https://docs.rs/tokio/latest/tokio/time/fn.timeout.html instead?

Reply:
if not, don't worry

Reply:
I don't quite understand how to use that in a select! block, maybe you can show me next week.

Reply:
Yeah course. My hope for using the timeout type itself is to eliminate the select block. Because it seems like you're only selecting between the stream and the timeout